### ISSUE TITLE
Leaflet js marker not showing in safari browser

### ISSUE THREAD

**Issue by yashweb3** [Leaflet/Leaflet#6644]
![image](https://user-images.githubusercontent.com/48712509/57369301-d432f880-71aa-11e9-8f3a-04bd47788e7f.png)

iOS devices and safari browser not showing markers.

**Comment by IvanSanchez** [comment]
Hi, I'm closing this issue because the quality is not good enough. I suggest reading ["How To Ask Questions The Smart Way"](http://www.catb.org/esr/faqs/smart-questions.html) and ["How to Report Bugs Effectively"](http://www.chiark.greenend.org.uk/~sgtatham/bugs.html) so you can make good bug reports.

**Comment by yashweb3** [comment]
I am getting a problem with the Safari browser. I am not able to show tracker in safari browser.

can you please help me with that

**Comment by IvanSanchez** [comment]
What problem? In what version of Safari? What's the code look like? What's the *minimal* code example that reproduces the issue?

> can you please help me with that?

With the amount of information you're providing, no.

**Comment by yashweb3** [comment]
_1. Library version and library name_

	Library name: leaflet.js
	Version:0.7.2 
	http://leafletjs.com
			 (c) 2010-2013, Vladimir Agafonkin
			 (c) 2010-2011, CloudMade

_2. Specific code where we are facing this issue and highlight what exactly not working in safari while other things working well in other browsers_

_layerAdd function running in ios Devices and safari browser

        ```
    _layerAdd: function(t) {
                console.log(t);
                t.onAdd(this), this.fire("layeradd", {
                    layer: t
                })
            },
```

            **but OnAdd function is not run on safari browser and ios devices.**

           ```
 onAdd: function(t) {
                this._map = t, t.on("viewreset", this.update, this), this._initIcon(), this.update(), this.fire("add"), t.options.zoomAnimation && t.options.markerZoomAnimation && t.on("zoomanim", this._animateZoom, this)
            },
```

            **so that's why _initIcon function also not running in ios devices and safari browser**

             ```
_initIcon: function() {
                console.log('call initIcon here');
                var t = this.options,
                    e = this._map,
                    i = e.options.zoomAnimation && e.options.markerZoomAnimation,
                    n = i ? "leaflet-zoom-animated" : "leaflet-zoom-hide",
                    s = t.icon.createIcon(this._icon),
                    a = !1;
                s !== this._icon && (this._icon && this._removeIcon(), a = !0, t.title && (s.title = t.title), t.alt && (s.alt = t.alt)), o.DomUtil.addClass(s, n), t.keyboard && (s.tabIndex = "0"), this._icon = s, this._initInteraction(), t.riseOnHover && o.DomEvent.on(s, "mouseover", this._bringToFront, this).on(s, "mouseout", this._resetZIndex, this);
                var r = t.icon.createShadow(this._shadow),
                    h = !1;
                r !== this._shadow && (this._removeShadow(), h = !0), r && o.DomUtil.addClass(r, n), this._shadow = r, t.opacity < 1 && this._updateOpacity();
                var l = this._map._panes;
                a && l.markerPane.appendChild(this._icon), r && h && l.shadowPane.appendChild(this._shadow)
            },
```

            **That's why marker is not showing in safari browser and ios device. in android devices and chrome and mozilla code leaflet library working fine.**

**Comment by yashweb3** [comment]
**1. Library version and library name**

	Library name: leaflet.js
	Version:0.7.2 
	http://leafletjs.com
			 (c) 2010-2013, Vladimir Agafonkin
			 (c) 2010-2011, CloudMade

**2. Specific code where we are facing this issue and highlight what exactly not working in safari while other things working well in other browser**

### _layerAdd function running in ios Devices and safari browser

`_layerAdd: function(t) {
                console.log(t);
                t.onAdd(this), this.fire("layeradd", {
                    layer: t
                })
            },
`
### but OnAdd function is not run on safari browser and ios devices.

`onAdd: function(t) {
                this._map = t, t.on("viewreset", this.update, this), this._initIcon(), this.update(), this.fire("add"), t.options.zoomAnimation && t.options.markerZoomAnimation && t.on("zoomanim", this._animateZoom, this)
            },
`
### so that's why _initIcon function also not running in ios devices and safari browser

`_initIcon: function() {
                console.log('call initIcon here');
                var t = this.options,
                    e = this._map,
                    i = e.options.zoomAnimation && e.options.markerZoomAnimation,
                    n = i ? "leaflet-zoom-animated" : "leaflet-zoom-hide",
                    s = t.icon.createIcon(this._icon),
                    a = !1;
                s !== this._icon && (this._icon && this._removeIcon(), a = !0, t.title && (s.title = t.title), t.alt && (s.alt = t.alt)), o.DomUtil.addClass(s, n), t.keyboard && (s.tabIndex = "0"), this._icon = s, this._initInteraction(), t.riseOnHover && o.DomEvent.on(s, "mouseover", this._bringToFront, this).on(s, "mouseout", this._resetZIndex, this);
                var r = t.icon.createShadow(this._shadow),
                    h = !1;
                r !== this._shadow && (this._removeShadow(), h = !0), r && o.DomUtil.addClass(r, n), this._shadow = r, t.opacity < 1 && this._updateOpacity();
                var l = this._map._panes;
                a && l.markerPane.appendChild(this._icon), r && h && l.shadowPane.appendChild(this._shadow)
            },
`

### That's why marker is not showing in safari browser and ios device. in android devices and chrome and Mozilla code leaflet library working fine.

**Comment by IvanSanchez** [comment]
We don't fix bugs on 0.7.x anymore.
